processor/manager.go

```diff
@@ -57,7 +58,10 @@ func (proc *LifecycleManager) Start() error {
 
 	var wg sync.WaitGroup
 	proc.waitGroup = &wg
-
+	metric.Instance.Reset()
+	if err := proc.Handle.countPendingEvents(); err != nil {
```

Comment:
I feel the whole start and stop needs to be refactored out since here the proc.Handle.Start is a blocking call and ideally Start shouldn't be a blocking call. I just felt that out for the future refactoring which we might take up